src/battleship/org/Battlefield.java

```diff
@@ -33,58 +40,88 @@ public void placeShip(Ship ship, String[] coordinates) {
+            for (int i = start[0]; i <= end[0]; i++) {
+                battleField[i][start[1]] = Designations.CELL;
+                locationOfShip.add(new Coordinates(i, start[1], Designations.CELL));
+            }
+        }
+        fleet.add(new Ship(shipClass, locationOfShip));
     }
 
-    public String markShot(String coordinates) {
-        var x = coordinates.charAt(0) - 65;
+    protected Mark markShot(String coordinates) {
+        var x = fromCharToInt(coordinates.charAt(0));
         var y = Integer.parseInt(coordinates.substring(1)) - 1;
 
         if (x >= Constant.FIELD_SIZE || x < 0 ||
                 y >= Constant.FIELD_SIZE || y < 0) {
             throw new IllegalArgumentException("Error! You entered the wrong coordinates! Try again:");
         }
-        if (Designations.CELL.equals(battleField[x][y])) {
+        mark(x, y, Designations.HIT);
+        if (Designations.CELL.equals(battleField[x][y]) || Designations.HIT.equals(battleField[x][y])) {
             battleField[x][y] = Designations.HIT;
-            return "You hit a ship!";
+            if (!isShipAlive(x, y) && checkIsAliveFleet()) {
+                return Mark.GAME_CONTINUOUS_SINK_SHIP;
+            }
+            return checkIsAliveFleet() ? Mark.GAME_CONTINUOUS : Mark.GAME_OVER;
         }
         battleField[x][y] = Designations.MISS;
-        return "You missed!";
+        return Mark.GAME_CONTINUOUS_MISS;
     }
 
-    private void checkNeighbors(int[] start, int[] end) {
-        var startPosX = (start[0] - 1 < 0) ? start[0] : start[0] - 1;
-        var startPosY = (start[1] - 1 < 0) ? start[1] : start[1] - 1;
-        var endPosX = (end[0] + 1 >= Constant.FIELD_SIZE) ? end[0] : end[0] + 1;
-        var endPosY = (end[1] + 1 >= Constant.FIELD_SIZE) ? end[1] : end[1] + 1;
+    private boolean isShipAlive(int x, int y) {
+        return !fleet.stream()
+                .filter(z -> z.exist(x, y))
+                .filter(Ship::isShipAlive)
+                .toList()
+                .isEmpty();
+    }
 
-        for (int a = startPosX; a <= endPosX; a++) {
-            for (int b = startPosY; b <= endPosY; b++) {
-                if (battleField[a][b] == Designations.CELL)
-                    throw new IllegalArgumentException("Error! You placed it too close to another one. Try again:");
+    private void mark(int x, int y, Designations designations) {
+        for (Ship ship : fleet) {
+            for (Coordinates coordinates : ship.getCompartments()) {
+                if (coordinates.getX() == x && coordinates.getY() == y) {
+                    coordinates.setStatus(designations);
+                }
             }
         }
     }
 
-    @Override
-    public String toString() {
+    public String displayPrimary() {
+        return draw(battleField, Designations.CELL);
+    }
+
+    public String displayEnemy() {
+        return displayShadow(battleField);
+    }
+
+    private String displayShadow(Designations[][] battleField) {
```

Comment:
Looks like `displayShadow` is not needed. As we can call `draw(battleField)` directly in `displayEnemy`